pkgs/development/compilers/yosys/fix-clang-build.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/Makefile b/Makefile
+index 86abc6958..a72f7b792 100644
+--- a/Makefile
++++ b/Makefile
+@@ -145,7 +145,12 @@ bumpversion:
+ ABCREV = 4f5f73d
+ ABCPULL = 1
+ ABCURL ?= https://github.com/YosysHQ/abc
++
++ifneq ($(CONFIG),clang)
++ABCMKARGS = CC=clang CXX="$(CXX)" LD=clang ABC_USE_LIBSTDCXX=1 VERBOSE=$(Q)
++else
+ ABCMKARGS = CC="$(CXX)" CXX="$(CXX)" ABC_USE_LIBSTDCXX=1 VERBOSE=$(Q)
++endif
```

Comment:
I don't understand this patch. Shouldn't this conditional be `ifeq`, rather than `ifneq`? Also, I don't see how this even fixes anything at all, because we use an external build of `abc` for Yosys (the `abc-verifier` package), that's matched to the version in the Makefile.

I assume the logic is that on Linux, `ABCMKARGS` is unused, so this helpfully does not actually cause a fatal error (since `clang` will not be in `$PATH`) while on Darwin something weird happens that means it's OK to use. But this shouldn't be happening this way...

Reply:
Yeah, that conditional is intended the other way, and given that it works nonetheless, touching `ABCMKARGS` probably doesn't do here anything as you suggest. (Building yosys with internal abc in a nix-shell is how I got to changing `ABCMKARGS`, but then I added the conditional wrong.)